src/test/java/com/cleancode/tdd/ReadingCounterTest.java

```diff
@@ -0,0 +1,55 @@
+package com.cleancode.tdd;
+
+
+
+import org.junit.Test;
+
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+public class ReadingCounterTest {
+
+    ReadingCounter readingCounter = new ReadingCounter();
+
+    @Test
+    public void readingCountOfGivenRangeTest_invalid_startRange() throws Exception{
```

Comment:
There is scope to improve the test cases. The input range is fixed and also part of production code instead of test code.
Various combination of input values need to be considered. Single valued, multivalued input array.

Also , the for a given array {1,2,3,5,6,8,9} the output should be 1-3,3  5-6,2  8-9,2
i.e. the range should be discovered and the elements count within that range needs to be found out.